### PR TITLE
Update DeepSeek-OCR: HF-direct loader with transformers 4.46.3 pin and hub patches

### PR DESCRIPTION
**Pull request by ashokkumarkannan1**
### Ticket
-  https://github.com/tenstorrent/tt-xla/issues/4835

### Problem description
- DeepSeek-OCR was originally built and tested on transformers==4.46.3 (see deepseek-ai/DeepSeek-OCR). tt-xla’s dev environment has since moved to newer Transformers (e.g. 5.2.0).
- To keep the model runnable, [#209](https://github.com/tenstorrent/tt-forge-models/pull/209) vendored and rewrote the full modeling_* stack locally (CUDA removal, LlamaFlashAttention2 / RoPE / attention API changes for ~4.52.4, easydict import fixes, etc.). That worked for bringup at the time but created a fork that drifts from upstream whenever Transformers changes.
- On 5.2.0, the forked path showed all-NaN logits on CPU (documented in [tt-xla#4703](https://github.com/tenstorrent/tt-xla/issues/4703)). A follow-up in forge restores finite logits on 5.2 by patching the local modeling_deepseekv2.py again ([5.2.0 fix](https://github.com/tenstorrent/tt-forge-models/compare/main...akannan/deepseek_nan_fix_clean))—but each Transformers bump risks new silent breakage (API shape, attention returns, numerics).

- Goal: stop maintaining a full local clone of hub modeling_*. Load deepseek-ai/DeepSeek-OCR from Hugging Face on the official 4.46.3 pin, let tt-xla RequirementsManager install/rollback that pin per test, and apply only small, documented hub text patches needed for CPU-only PyTorch and tt-xla compile.
- Known limitation (unchanged by this PR): after patches, inference runs, but tt-xla vs CPU golden PCC remains ~0.94 (threshold 0.99)—same ballpark as the 5.2 NaN-fix path. This PR fixes loader architecture and reproducibility, not device numerics. that PCC Drop to be investigated later.

### What's changed
HF-direct loader (deepseek/deepseek_ocr/pytorch/loader.py):
- snapshot_download of deepseek-ai/DeepSeek-OCR into DeepSeek_OCR_hub/ at a pinned revision (DEFAULT_HF_REVISION, overridable via DEEPSEEK_OCR_HF_REVISION).
- AutoModel.from_pretrained(..., trust_remote_code=True, local_files_only=True) on the patched snapshot (no local modeling_* fork).

- Removed local neural-net fork (~2.2k lines): Deleted src/modeling_deepseekocr.py, src/modeling_deepseekv2.py, src/deepencoder.py, src/configuration_deepseek_v2.py.
- New src/model.py — hub runtime only:
     a) Patch 1 — masked_scatter: masked_scatter_ + .cuda() → masked_scatter on CPU (tt-xla masked_scatter decomposition path; same intent as #209).
    b) Patch 2 — torch.sum: explicit dim=(0,1,2,3) for image non-empty check (XLA-friendly reduction).
    c) Patch 3 — MoE: tokens_per_expert.cpu().numpy() → .cpu().tolist() (avoids Dynamo 'ndarray' has no attribute 'dim').

- CPU CUDA workaround: no-op Tensor.cuda() when PyTorch has no CUDA (hub still calls .cuda() in places).
- invalidate_transformers_hub_module_cache(): clear HF dynamic-module cache so patched hub files are actually loaded.
- src/model_utils.py — input preprocessing only (tokenizer + image tiling from hub conversation.py); dead code trimmed.
- Per-model requirements.txt: transformers==4.46.3, addict, easydict (hub remote code deps; not in tt-xla root requirements).

### Checklist
- [x] New/Existing tests provide coverage for changes

### Logs
- [deepseek_pcc_after_bringup.log](https://github.com/user-attachments/files/28112383/deepseek_pcc_after_bringup.log)

### CODE REVIEW

**Review by kamalrajkannan78 — APPROVED** [review]

File: deepseek/deepseek_ocr/pytorch/requirements.txt [hunk 1]
```diff
@@ -1 +1,3 @@
+addict>=2.4.0
 easydict==1.13
+transformers==4.46.3 # https://huggingface.co/deepseek-ai/DeepSeek-OCR/blob/main/config.json#L62
```
[the remark this answers is not in the document]
Comment (ppadjinTT) [reply deepseek/deepseek_ocr/pytorch/requirements.txt:3]: similar to this [comment](https://github.com/tenstorrent/tt-forge-models/pull/678#discussion_r3287005346), not sure if this is okay

**Review by ppadjinTT — COMMENTED** [review]
let's pause untill we discuss this requirements fix

File: deepseek/deepseek_ocr/pytorch/requirements.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashokkumarkannan1) [reply deepseek/deepseek_ocr/pytorch/requirements.txt:3]: @ppadjinTT 

 ### Why transformers==4.46.3 (not tt-xla’s 5.2.0 pin)

- Official model pin — deepseek-ai/DeepSeek-OCR targets 4.46.3 in hub [config.json](https://huggingface.co/deepseek-ai/DeepSeek-OCR/blob/main/config.json#L62); we load hub remote code on that version, not a downgraded guess.
- Avoid re-fixing on every tt-xla bump: 
  a) The old vendored fork had to be patched again for ~4.52.4 - https://github.com/tenstorrent/tt-forge-models/pull/209.
 b) on 5.2.0 the fork path hit all-NaN logits ([tt-xla#4703](https://github.com/tenstorrent/tt-xla/issues/4703), fixed in [deepseek_nan_fix_clean](https://github.com/tenstorrent/tt-forge-models/compare/main...akannan/deepseek_nan_fix_clean))

### Per-model requirements.txt + HF-direct loader isolates this model from global Transformers upgrades.

### Same approach as [PI0 Model (Transformers 4.53.3)](https://github.com/tenstorrent/tt-forge-models/blob/main/pi_0/pytorch/requirements.txt#L1), which pins a model-specific Transformers build (patched git ref) because bringup depends on that stack

**Review by ppadjinTT — CHANGES_REQUESTED** [review]
same comment as [here](https://github.com/tenstorrent/tt-forge-models/pull/678#pullrequestreview-4345703661), better to mark this model as xfail then to add per-model hacks.
